search.go

```diff
@@ -125,20 +126,22 @@ func NewTimeRange(endTime, period int64) (*TimeRange, error) {
 
 // Execute is used to carry out a search
 func (s *Search) Execute() (*SearchResponse, error) {
+	paramsCopy := *s.Params
```

Comment:
So the conditions is still modifiable since its passed by reference.  Should we worry about that?

Reply:
Good point. You are right. If they modify the search conditions in place, the copy by assignment that I did won't help.  However, in GO it is understood that it is best to make changes to structs with all public fields in a way that is compatible with assignment. Otherwise, you are asking for trouble.

Reply:
Perhaps we should add deep copy in NewSearch()?

Reply:
I actually didn't look closely at which line you were commenting on.  My change on line 129 is enough because all we need to do is ensure that Execute doesn't change internal state of `s`. 

As far as doing a deep copy on NewSearch, it is hard to get right.  For example, I will be adding a `values []string` field to SearchCondition because now the REST API lets you search over multiple values.  Right now, copy by assignment is enough for a SearchCondition, but with this new field I would need to remember to deep copy even SearchCondition.

Reply:
alright, that's fine.  lgtm